python/src/iceberg/expressions/base.py

```diff
@@ -138,6 +75,24 @@ def __invert__(self) -> "BooleanExpression":
         ...
 
 
+class BoundPredicate(ABC):
+    def __init__(self, term, literal):
+        """A concrete predicate must have a `term` and `literal`"""
+        self._term = term
+        self._literal = literal
+
+
+class UnboundPredicate:
```

Comment:
I think this and `BoundPredicate` should extend `BooleanExpression` because they return boolean values.

Also, there should be no need for `BoundPredicate` to extend ABC right now, since there are no methods defined for it that aren't defined.